modules/xmpp/JingleSessionPC.js

```diff
@@ -307,118 +278,15 @@ JingleSessionPC.prototype.acceptOffer = function(jingleOffer,
-    this.readSsrcInfo($(jingleOfferIq).find(">content"));
-    var remotedesc
-        = new RTCSessionDescription({type: 'offer', sdp: this.remoteSDP.raw});
-
-    this.peerconnection.setRemoteDescription(remotedesc,
-        function () {
-            //logger.log('setRemoteDescription success');
-            if (success) {
-                success();
-            }
-        },
-        function (e) {
-            logger.error('setRemoteDescription error', e);
-            if (failure)
-                failure(e);
-            JingleSessionPC.onJingleFatalError(this, e);
-        }.bind(this)
-    );
-};
-
-/**
- * This is a wrapper to PeerConnection.createAnswer in order to generate failure
- * event when error occurs. It also includes "media_constraints" if any are set
- * on this JingleSessionPC instance.
- * @param success callback called when PC.createAnswer succeeds, SDP will be
- *        the first argument
- * @param failure callback called with error argument when setAnswer fails
- */
-JingleSessionPC.prototype.createAnswer = function (success, failure) {
-    //logger.log('createAnswer');
-    var self = this;
-    this.peerconnection.createAnswer(
-        function (answer) {
-            var modifiedAnswer = new SDP(answer.sdp);
-            JingleSessionPC._fixAnswerRFC4145Setup(
-                /* offer */ self.remoteSDP,
-                /* answer */ modifiedAnswer);
-            answer.sdp = modifiedAnswer.raw;
-            success(answer);
-        },
-        function (error) {
-            logger.error("createAnswer failed", error);
-            if (failure)
-                failure(error);
-            self.room.eventEmitter.emit(
-                    XMPPEvents.CONFERENCE_SETUP_FAILED, error);
-        },
-        this.media_constraints
-    );
-};
-
-JingleSessionPC.prototype.setLocalDescription = function (sdp, success,
-                                                               failure) {
-    var self = this;
-    this.localSDP = new SDP(sdp.sdp);
-    sdp.sdp = this.localSDP.raw;
-    this.peerconnection.setLocalDescription(sdp,
-        function () {
-            if (success)
-                success();
-        },
-        function (error) {
-            logger.error('setLocalDescription failed', error);
-            if (failure)
-                failure(error);
-            self.room.eventEmitter.emit(XMPPEvents.CONFERENCE_SETUP_FAILED);
-        }
-    );
-    // Some checks for STUN and TURN candiates present in local SDP
-    // Eventually could be removed as we don't really care
-    var cands = SDPUtil.find_lines(this.localSDP.raw, 'a=candidate:');
-    for (var j = 0; j < cands.length; j++) {
-        var cand = SDPUtil.parse_icecandidate(cands[j]);
-        if (cand.type == 'srflx') {
-            this.hadstuncandidate = true;
-        } else if (cand.type == 'relay') {
-            this.hadturncandidate = true;
-        }
-    }
+    this.jingleOfferIq = jingleOfferIq;
```

Comment:
I agree that having SRD/SLD cycle on the queue is a good thing, but just don't like how it is and was implemented. In my opinion that is not very readable, because mixes many different operations into one method which come from different places around the code. Agree that we can fix that another time, but I'm just not a big fan of piling onto hacky solutions instead of fixing them asap.